tests/lvmlib.py

```diff
@@ -72,51 +72,53 @@ def fake_lvchange(self, args, stdin):
 
     def fake_lvcreate(self, args, stdin):
         self.logger('lvcreate', repr(args), stdin)
-        parser = TestOptParse()
-        parser.add_option("-n", dest='name')
-        parser.add_option("-L", dest='size_mb')
-        parser.add_option("--addtag", dest='tag')
-        parser.add_option("--inactive", dest='inactive', action='store_true')
-        parser.add_option("--zero", dest='zero', default='y')
-        parser.add_option("-W", dest='wipe_sig')
+        parser = TestArgParse(prog='lvcreate')
+        parser.add_argument("-n", dest='name')
+        parser.add_argument("-L", dest='size_mb')
+        parser.add_argument("--addtag", dest='tag')
+        parser.add_argument("--inactive", dest='inactive', action='store_true')
+        parser.add_argument("--zero", dest='zero', default='y')
+        parser.add_argument("-W", dest='wipe_sig')
+        parser.add_argument('vgname')
         try:
-            options, args = parser.parse_args(args[1:])
+            args = parser.parse_args(args[1:])
         except SystemExit as e:
             self.logger("LVCREATE OPTION PARSING FAILED")
             return (1, '', str(e))
 
-        vgname, = args
+        vgname = args.vgname
 
         if self.get_volume_group(vgname) is None:
             self.logger("volume group does not exist:", vgname)
             return (1, '', '  Volume group "%s" not found\n' % vgname)
 
-        active = not options.inactive
-        assert options.zero in ['y', 'n']
-        zeroed = options.zero == 'y'
+        active = not args.inactive
+        assert args.zero in ['y', 'n']
+        zeroed = args.zero == 'y'
 
         self.get_volume_group(vgname).add_volume(
-            options.name,
-            int(options.size_mb),
-            options.tag,
+            args.name,
+            int(args.size_mb),
```

Comment:
Not that it matters particularly but it might be preferable to `parser.add_argument("-L", dest='size_mb', type=int)` instead.

Reply:
Yes, sensible, it was a basic search replace for the most part but that would be a further improvement in maintainability here.